op-node/rollup/derive/channel_bank.go

```diff
@@ -81,85 +81,73 @@ func (ib *ChannelBank) IngestData(data []byte) error {
 	if data[0] != DerivationVersion0 {
 		return fmt.Errorf("unrecognized derivation version: %d", data)
 	}
+	var buf bytes.Buffer
+	buf.Write(data[1:])
```

Comment:
Maybe just avoid copying the complete input with something like `input := bytes.NewReader(data[1:])`? This provides the same byte-reading methods as the bytes buffer, without copy.